src/features/room/helpers/join.tsx

```diff
@@ -116,10 +125,103 @@ export default function join(
             const ok = yield* recover(
                 function* () {
                     try {
-                        const tx: Record<string, any> = yield policy.requestDelegatedJoin()
+                        if (stakingEnabled) {
+                            yield onToast({
+                                title: i18n('joinTokenGatedRoomToast.authorizeTokenTransfer'),
+                                type: ToastType.Processing,
+                            })
+                            let authorizationTx: Record<string, any>
+                            let tokenContract: Contract
+                            switch (tokenStandard) {
+                                case TokenStandard.ERC20:
+                                    tokenContract = new Contract(
+                                        tokenAddress,
+                                        ERC20TokenAbi,
+                                        policyRegistry.signer
+                                    )
+                                    authorizationTx = yield tokenContract.approve(
+                                        policyAddress,
+                                        BigNumber.from(minRequiredBalance || 0)
+                                    )
+                                    break
+                                case TokenStandard.ERC721:
+                                    tokenContract = new Contract(
+                                        tokenAddress,
+                                        ERC721TokenAbi,
+                                        policyRegistry.signer
+                                    )
+                                    authorizationTx = yield tokenContract.approve(
+                                        policyAddress,
+                                        BigNumber.from(tokenId)
+                                    )
+                                    break
+                                default:
+                                    throw new Error(`Unsupported token standard ${tokenStandard}`)
+                            }
 
+                            yield authorizationTx.wait()
+                            yield onToast({
+                                title: i18n('joinTokenGatedRoomToast.authorizationComplete'),
+                                type: ToastType.Success,
+                            })
```

Comment:
This toast will never be seen. Another `onToast` happens right after it (see below). Let's maybe have a separate toast instance for the spending authorization.